napari/layers/surface/surface.py

```diff
@@ -0,0 +1,309 @@
+from typing import Union
+from xml.etree.ElementTree import Element
+import numpy as np
+from copy import copy
+import vispy.color
+from ..base import Layer
+from ...util.event import Event
+from ..image._constants import Rendering, Interpolation, AVAILABLE_COLORMAPS
+from ...util.status_messages import format_float
+from ...util.misc import calc_data_range, increment_unnamed_colormap
+from vispy.color import get_color_names, Color
+
+
+class Surface(Layer):
+    """
+    Surface layer renders meshes onto the canvas.
+
+    Parameters
+    ----------
+    data : (N, D) array
+        Vertices of mesh triangles.
+    faces : (M, 3) array of int
+        Indices of mesh triangles.
+    values : (N,) array
+        Values used to color vertices.
+    colormap : str, vispy.Color.Colormap, tuple, dict
+        Colormap to use for luminance images. If a string must be the name
+        of a supported colormap from vispy or matplotlib. If a tuple the
+        first value must be a string to assign as a name to a colormap and
+        the second item must be a Colormap. If a dict the key must be a
+        string to assign as a name to a colormap and the value must be a
+        Colormap.
+    contrast_limits : list (2,)
+        Color limits to be used for determining the colormap bounds for
+        luminance images. If not passed is calculated as the min and max of
+        the image.
+    name : str
+        Name of the layer.
+    metadata : dict
+        Layer metadata.
+    scale : tuple of float
+        Scale factors for the layer.
+    translate : tuple of float
+        Translation values for the layer.
+    opacity : float
+        Opacity of the layer visual, between 0.0 and 1.0.
+    blending : str
+        One of a list of preset blending modes that determines how RGB and
+        alpha values of the layer visual get mixed. Allowed values are
+        {'opaque', 'translucent', and 'additive'}.
+    visible : bool
+        Whether the layer visual is currently being displayed.
```

Comment:
Do we have a concept of lighting in this layer? (ie if you render something all the same colour, is it going to be visible as anything other than a homogeneous blob?) And if so, do we want to be able to control it?

Reply:
there are some lighting controls easily accessible with the vispy mesh object - see here http://vispy.org/visuals.html#vispy.visuals.MeshVisual - but I'm not sure about how much we want to expose. I worry about too much customizability and too many options

Reply:
Sure, absolutely, but what is it currently set to? I guess it should at least be documented somewhere what the default is.

Reply:
i've added a comment on the `VispySurfaceLayer` class